lib/puppet/module.rb

```diff
@@ -305,6 +305,16 @@ def locale_directory
     subpath("locales")
   end
 
+  # Returns true if the module has translation files for the
+  # given locale.
+  # @param [String] locale the two-letter language code to check
+  #        for translations
+  # @return true if the module has a directory for the locale, false
+  #         false otherwise
+  def has_translations?(locale)
+    return Puppet::FileSystem.exist?(File.join(locale_directory, locale))
```

Comment:
This feels potentially over-restrictive; I'm not sure what algorithm gettext uses to find locale directories based on a locale, but it might not be a direct mapping. However, we control the ecosystem so it's probably fine if we're more restrictive.

Reply:
So FastGettext allows any folder that [matches](https://github.com/grosser/fast_gettext/blob/master/lib/fast_gettext/translation_repository/base.rb#L41-L53) this regex: https://github.com/grosser/fast_gettext/blob/fa8da8d6947108955450b1d0b515e32a36ee5c38/lib/fast_gettext.rb#L12 However, when it goes to reference the files it loads, it only uses the entry for the current locale: https://github.com/grosser/fast_gettext/blob/fa8da8d6947108955450b1d0b515e32a36ee5c38/lib/fast_gettext/translation_repository/mo.rb#L41 So I think you're right, if that's the only way we are using it (i.e. we do not allow the locale to change per thread or at runtime in any way), I think this is safe.

Alternatively, we could more generally ask if there are any subdirectories, but then I think if we only have languages we're not using, we'll still do the work of checking the empty MO file.